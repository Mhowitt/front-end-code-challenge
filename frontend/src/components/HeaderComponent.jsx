import React from 'react';
import { Link } from 'react-router-dom';

const HeaderComponent = props => {
  return (
    <div style={{width: '100%', height:'50px', backgroundColor: 'lavender'}}>
      Clothing Helper

    </div>
  )
}

export default HeaderComponent;