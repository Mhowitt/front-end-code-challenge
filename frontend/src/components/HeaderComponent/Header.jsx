import React from 'react';
import { Link } from 'react-router-dom';
import './Header.css';

const Header = props => {
  return (
    <div className="header-container">
      <div className="title-header">
        <Link to={'/'} onClick={props.goHome}>
          RightFit
        </Link>
      </div>
      {props.takingQuiz ? (
        ''
      ) : (
        <div className="button-header">
          <Link to={'/profile'}>
            <button className="start-quiz" onClick={props.startQuiz}>
              Take our Quiz
            </button>
          </Link>
        </div>
      )}
    </div>
  );
};

export default Header;
