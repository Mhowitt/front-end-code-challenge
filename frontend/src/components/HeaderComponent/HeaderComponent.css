.header-container {
  width: 100%;
  height:10vh;
  background-color: lavender;
  color: #6D7993;
  position: relative;
  display: flex;
  flex-direction: row;
  align-items: center;
}
a {
  text-decoration: none;
  color: #6D7993;
}
.title-header {
  font-size: 22px;
  margin-left: 1%;
}


.button-header {
  position: absolute;
  right: 1%;
  align-self: center;
}