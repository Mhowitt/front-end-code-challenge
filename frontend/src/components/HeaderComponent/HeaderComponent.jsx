import React from 'react';
import { Link } from 'react-router-dom';
import './HeaderComponent.css';

const HeaderComponent = props => {
  return (
    <div className="header-container">
      <div className="title-header">
        <Link to={'/'} onClick={props.goHome}>
          RightFit
        </Link>
      </div>
      {props.takingQuiz ? (
        ''
      ) : (
        <div className="button-header">
          <Link to={'/profile'}>
            <button className="start-quiz" onClick={props.goHome}>
              Take our Quiz
            </button>
          </Link>
        </div>
      )}
    </div>
  );
};

export default HeaderComponent;
