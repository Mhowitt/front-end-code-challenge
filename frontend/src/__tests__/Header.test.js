import React from 'react'; // for JSX
import { shallow, mount } from 'enzyme'; // how to mount the component
import toJson from 'enzyme-to-json';
import HeaderComponent from '../components/HeaderComponent/Header.jsx';

describe('<HeaderComponent />', () => {
  let wrapper;
  it('renders', () => {
    wrapper = shallow(<HeaderComponent />);
  });
  it('matches snapshot', () => {
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
