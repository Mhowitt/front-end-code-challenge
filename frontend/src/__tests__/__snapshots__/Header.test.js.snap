// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderComponent /> matches snapshot 1`] = `
<div
  className="header-container"
>
  <div
    className="title-header"
  >
    <Link
      replace={false}
      to="/"
    >
      RightFit
    </Link>
  </div>
  <div
    className="button-header"
  >
    <Link
      replace={false}
      to="/profile"
    >
      <button
        className="start-quiz"
      >
        Take our Quiz
      </button>
    </Link>
  </div>
</div>
`;
